pkg/p2p/p2p.go

```diff
@@ -64,7 +73,42 @@ func standardHostOpts(priv crypto.PrivKey, listenAddrs []ma.Multiaddr) []config.
 
 }
 
-func LibP2PHTTPRequestWithRandomClient(addr ma.Multiaddr, route string, peerID peer.ID, req *http.Request) (*http.Response, error) {
+func PostLibP2PRequestToAddress(node *Node, proxyAddr string, route string, req *http.Request) ([]byte, error) {
```

Comment:
I think this might be confusing because there is a function with the same name in the `habitat_client` package. It might make more sense to create a method on the `Node` struct in the node package that makes a LibP2P http request